File: backend/api/src/main/java/kpn/api/common/monitor/MonitorRouteSummary.java
package kpn.api.common.monitor;

import kpn.api.common.Bounds;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record MonitorRouteSummary(
  Boolean adminUser,
  String groupName,
  String routeName,
  String routeDescription,
  String routeId,
  Optional<Long> relationId,
  ImmutableList<Long> relationIds,
  Long memberCount,
  Long segmentCount,
  Long deviationCount,
  Optional<Bounds> bounds
) {}

File: backend/api/src/main/scala/kpn/api/common/monitor/MonitorRouteSummary.scala
package kpn.api.common.monitor

import kpn.api.common.Bounds

case class MonitorRouteSummary(
  adminUser: Boolean,
  groupName: String,
  routeName: String,
  routeDescription: String,
  routeId: String,
  relationId: Option[Long],
  relationIds: Seq[Long],
  memberCount: Long,
  segmentCount: Long,
  deviationCount: Long,
  bounds: Option[Bounds]
)
